src/prebase.ts

```diff
@@ -153,11 +154,18 @@ export class ProtoM21Object {
                     clonedVersion = m21Obj.clone(deep, memo);
                 }
                 ret[key] = clonedVersion;
+            } else if (
+                deep
+                && this[key] instanceof Array
+            ) {
+                ret[key] = Array.from(this[key] as any);
             } else {
                 try {
-                    // for deep this should be:
-                    // music21.common.merge(ret[key], this[key]);
-                    ret[key] = this[key];
+                    if (deep) {
+                        common.merge(ret[key], this[key] as any);
```

Comment:
I'm guessing this is not enough.  Probably need to do something more like:

Split the section beginning:
```for (const key in this) {```
into its own method (common.cloneHelper?) (i.e. a function that doesn't use "this" but `obj` or something like that).  And then when encountering another dictionary/object, calling cloneHelper on that.